sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverClientIntegrationTest.java

```diff
@@ -0,0 +1,578 @@
+    final AtomicReference<List<Long>>  messagesDeferred = new AtomicReference<>(new ArrayList<>());
+
+
+    protected ServiceBusReceiverClientIntegrationTest() {
+        super(new ClientLogger(ServiceBusReceiverClientIntegrationTest.class));
+    }
+
+    @Override
+    protected void beforeTest() {
+        sessionId = UUID.randomUUID().toString();
+    }
+
+    @Override
+    protected void afterTest() {
+        final int pending = messagesPending.get();
+        if (pending < 1 && messagesDeferred.get().size() < 1) {
+            dispose(receiver, sender, receiveAndDeleteReceiver);
+            return;
+        }
+        // In the case that this test failed... we're going to drain the queue or subscription.
+        if (pending > 0) {
+            try {
+                IterableStream<ServiceBusReceivedMessage> removedMessage = receiveAndDeleteReceiver.receive(
+                    pending + BUFFER_MESSAGES_TO_REMOVE);
+                removedMessage.stream().forEach(receivedMessage -> {
+                    System.out.println("Removed Message sequence : " +  receivedMessage.getSequenceNumber());
+                });
+            } catch (Exception e) {
+                logger.warning("Error occurred when draining queue.", e);
+            }
+        }
+
+        if (messagesDeferred.get().size() > 0) {
+            try {
+                List<Long> listOfDeferredMessages =  messagesDeferred.get();
+                for (Long seqNumber : listOfDeferredMessages) {
+                    ServiceBusReceivedMessage deferredMessages = receiver.receiveDeferredMessage(seqNumber);
+                    receiver.complete(deferredMessages);
+                }
+            } catch (Exception e) {
+                logger.warning("Error occurred when draining deferred messages Entity : " + receiver.getEntityPath(), e);
+            }
+
+            // set empty list
+            messagesDeferred.set(new ArrayList<>());
+        }
+
+        dispose(receiver, sender, receiveAndDeleteReceiver);
+    }
+
+    /**
+     * Verifies that we can send and receive two messages.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveTwoMessagesAutoComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        int howManyMessage = 2;
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+
+        sendMessage(message);
+        sendMessage(message);
+
+        // Act
+        Iterable<ServiceBusReceivedMessage> iterableMessages = receiver.receive(howManyMessage, TIMEOUT);
+
+        // Assert
+        for (ServiceBusReceivedMessage receivedMessage: iterableMessages) {
+            assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+        }
+
+        messagesPending.decrementAndGet();
```

Comment:
The reason this is here is that when we successfully receive a message, we don't need to clean it up in after test case scenario. This should be in the foreach block. Same with any uses below.